cs/src/core/Allocator/PendingFlushList.cs

```diff
@@ -1,54 +1,43 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license.
 
-using System.Threading;
+using System.Collections.Generic;
 
 namespace FASTER.core
 {
-    class PendingFlushList
+    sealed class PendingFlushList
     {
-        const int maxSize = 8;
-        const int maxRetries = 10;
-        public PageAsyncFlushResult<Empty>[] list;
+        public readonly LinkedList<PageAsyncFlushResult<Empty>> list;
 
         public PendingFlushList()
         {
-            list = new PageAsyncFlushResult<Empty>[maxSize];
+            list = new();
         }
 
-        public bool Add(PageAsyncFlushResult<Empty> t)
+        public void Add(PageAsyncFlushResult<Empty> t)
         {
-            int retries = 0;
-            do
+            lock (list)
             {
-                for (int i = 0; i < maxSize; i++)
-                {
-                    if (list[i] == default)
-                    {
-                        if (Interlocked.CompareExchange(ref list[i], t, default) == default)
-                        {
-                            return true;
-                        }
-                    }
-                }
-            } while (retries++ < maxRetries);
-            return false;
+                list.AddFirst(t);
```

Comment:
just checking: we basically don't have any ordering guarantees within the pending flush list right? 

Is there maybe some value in ensuring order? Normally, we don't flush a random range to disk, so only truly concurrent flushes will result in out-of-order entries in the pending flush list. In that case, whatever saving we have on the add path will arguably be negated on the remove path. What about just biting the bullet and ensure ordering from the start?

Reply:
Ensuring order would be useful. The entries are of the form [0, 23], [24, 39], [40, 100] and so on. But keeping these ordered will make this a O(log N) structure (interval tree).

Reply:
I was thinking about a much simpler approach: if you have [0, 23], and you see [40, 100] --- [24,39] must have been issued from a different thread, we can just spin wait until that's in before enqueuing [40, 100]